cloudfront/wellcomelibrary.org/edge-lambda/src/bnumberToWork.ts

```diff
@@ -0,0 +1,34 @@
+import { apiQuery, GetWorkResult, Identifier } from './api';
+
+export async function getWork(bNumber: string): Promise<GetWorkResult> {
+  const resultList = apiQuery({
+    query: bNumber,
+    include: ['identifiers'],
+  });
+
+  let work;
+
+  for await (const result of resultList) {
+    work = result;
+
+    if (result.identifiers) {
+      const identifiers: Identifier[] = result.identifiers;
+      // If the work has a sierra-identifier identifier, that
+      // preferentially identifiers the work as being sourced
+      // from Sierra, so use that work if we see it.
+      const filteredIdentifier = identifiers.filter(
+        (thing) => thing.identifierType.id === 'sierra-identifier'
+      );
+
+      if (filteredIdentifier.length >= 1) {
+        break;
+      }
```

Comment:
I think you can express this more directly:
```typescript
const hasSierraId = identifiers.some(
  thing => thing.identifierType.id === 'sierra-identifier'
);
if (hasSierraId) {
  break;
}
```

Reply:
In fact you might be able to rewrite this function without the mutation & break:
```typescript
const resultList = apiQuery({
  query: bNumber,
  include: ['identifiers'],
});

for await (const result of resultList) {
  const identifiers: Identifier[] = result.identifiers;
  const hasSierraId = identifiers.some(
    thing => thing.identifierType.id === 'sierra-identifier'
  );
  if (hasSierraId) {
    return result;
  }
}

return Error('No matching Catalogue API results found');
```

Reply:
I've a hard time trying to make this work without mutation while trying to keep the leniency about the existence of identifiers & fallback to record without matching a sierra identifier identifier. The use of `some` here though is a revelation.